kong/core/globalpatches.lua

```diff
@@ -124,9 +124,13 @@ return function(options)
         end
         return true
       end
-      function SharedDict:incr(key, value)
+      function SharedDict:incr(key, value, init)
         if not self.data[key] then
-          return nil, "not found"
+          if not init then
+            return nil, "not found"
+          else
+            self.data[key].value = init
+          end
```

Comment:
From https://github.com/openresty/lua-nginx-module#ngxshareddictincr:


> When the key does not exist or has already expired in the shared dictionary,
> 
> if the init argument is not specified or takes the value nil, this method will return nil and the error string "not found", or
> if the init argument takes a number value, this method will create a new key with the value init + value.

My understanding of reading the second line is the end result should equal `value + init`. Additionally:

https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_shdict.c#L1338
https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_shdict.c#L2667